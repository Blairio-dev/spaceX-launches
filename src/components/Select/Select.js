import PropTypes from "prop-types";
import React from "react";
import styled from "@emotion/styled";
import { Image } from "../../components";
import SelectIcon from "../../assets/icon/select@3x.png";
import SelectIcon2x from "../../assets/icon/select@2x.png";
import SelectIcon3x from "../../assets/icon/select@3x.png";
import { colours } from "../../assets/tokens";

const StyledSelect = styled("select")`
  appearance: none;
  background-color: ${colours.primaryBlue};
  border: 2px solid ${colours.primaryBlue};
  color: ${colours.white};
  cursor: pointer;
  font-family: "Brandon Grotesque Bold";
  font-size: 16px;
  height: 43px;
  padding: 0 37px 0 16px;

  :focus,
  :hover {
    border-color: ${colours.grey};
    outline: 0;
  }
`;

const StyledWrapper = styled("div")`
  align-items: center;
  color: ${colours.white};
  display: inline-flex;
  height: min-content;
  position: relative;
  width: fit-content;

  img {
    pointer-events: none;
    position: absolute;
    right: 24px;
    top: 19px;
  }
`;

const Select = ({ id, onChange, selectedOption, selectOptions }) => (
  <StyledWrapper>
    <StyledSelect id={id} onChange={onChange} value={selectedOption}>
      {selectOptions.map((item, index) => {
        const key = `${id}-${index}`;
        return (
          <option key={key} value={item}>
            {item}
          </option>
        );
      })}
    </StyledSelect>
    <Image
      alt="Chevron down"
      height="5px"
      sourceSet={`
			${SelectIcon3x} 3x,
			${SelectIcon2x} 2x,
			${SelectIcon} 1x,
		`}
    />
  </StyledWrapper>
);

Select.propTypes = {
  /** Unique string id. */
  id: PropTypes.string,
  /** Callback function triggered on change. */
  onChange: PropTypes.func,
  /** Selected option. */
  selectedOption: PropTypes.string.isRequired,
  /** Array of string options. */
  selectOptions: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export { Select };
