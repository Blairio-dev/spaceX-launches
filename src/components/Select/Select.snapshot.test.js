import React from "react";
import renderer from "react-test-renderer";
import { Select } from "./Select";

const options = ["1995", "1996", "1997", "1998", "1999", "2000"];

const DEFAULT_PROPS = {
  id: "test-select",
  onChange: jest.fn(),
  selectedOption: "",
  selectOptions: options,
};

describe("Select Snapshots", () => {
  it("Renders with default props", () => {
    const tree = renderer.create(<Select {...DEFAULT_PROPS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
