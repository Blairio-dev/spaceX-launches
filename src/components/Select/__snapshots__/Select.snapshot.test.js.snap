// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Snapshots Renders with default props 1`] = `
<div
  className="css-ovst58"
>
  <select
    className="css-1bjzmyt"
    id="test-select"
    onChange={[MockFunction]}
    value=""
  >
    <option
      value="1995"
    >
      1995
    </option>
    <option
      value="1996"
    >
      1996
    </option>
    <option
      value="1997"
    >
      1997
    </option>
    <option
      value="1998"
    >
      1998
    </option>
    <option
      value="1999"
    >
      1999
    </option>
    <option
      value="2000"
    >
      2000
    </option>
  </select>
  <img
    alt="Chevron down"
    height="5px"
    srcSet="
			select@3x.png 3x,
			select@2x.png 2x,
			select@3x.png 1x,
		"
  />
</div>
`;
